### PR TITLE
Telescope fzy native

### PR DESCRIPTION
**Pull request by onsails**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onsails** [comment]
Result of `nixpkgs-review pr 106592` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>spacevim</li>
    <li>vimPlugins.awesome-vim-colorschemes</li>
    <li>vimPlugins.coc-nvim</li>
    <li>vimPlugins.defx-nvim</li>
    <li>vimPlugins.nvim-highlite</li>
    <li>vimPlugins.nvim-tree-lua</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.plenary-nvim</li>
    <li>vimPlugins.telescope-fzy-native-nvim</li>
    <li>vimPlugins.telescope-nvim</li>
    <li>vimPlugins.vim-clap</li>
    <li>vimPlugins.vim-devicons</li>
    <li>vimPlugins.vim-terraform</li>
    <li>vimPlugins.vimtex</li>
  </ul>
</details>

**Comment by onsails** [comment]
Result of `nixpkgs-review pr 106592` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>spacevim</li>
    <li>vimPlugins.Tagbar (vimPlugins.tagbar)</li>
    <li>vimPlugins.ale</li>
    <li>vimPlugins.barbar-nvim</li>
    <li>vimPlugins.brainfuck-vim</li>
    <li>vimPlugins.coc-nvim</li>
    <li>vimPlugins.defx-nvim</li>
    <li>vimPlugins.denite (vimPlugins.denite-nvim)</li>
    <li>vimPlugins.deoplete-fish</li>
    <li>vimPlugins.deoplete-khard</li>
    <li>vimPlugins.deoplete-nvim</li>
    <li>vimPlugins.fern-vim</li>
    <li>vimPlugins.fzf-vim</li>
    <li>vimPlugins.julia-vim</li>
    <li>vimPlugins.lh-brackets</li>
    <li>vimPlugins.lh-vim-lib</li>
    <li>vimPlugins.nlua-nvim</li>
    <li>vimPlugins.nvim-gdb</li>
    <li>vimPlugins.nvim-lspconfig</li>
    <li>vimPlugins.nvim-lsputils</li>
    <li>vimPlugins.nvim-scrollview</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-refactor</li>
    <li>vimPlugins.nvim-ts-rainbow</li>
    <li>vimPlugins.popfix</li>
    <li>vimPlugins.snipmate (vimPlugins.vim-snipmate)</li>
    <li>vimPlugins.telescope-fzy-native-nvim</li>
    <li>vimPlugins.telescope-nvim</li>
    <li>vimPlugins.unite (vimPlugins.unite-vim)</li>
    <li>vimPlugins.vim-clap</li>
    <li>vimPlugins.vim-floaterm</li>
    <li>vimPlugins.vim-illuminate</li>
    <li>vimPlugins.vim-lsp</li>
    <li>vimPlugins.vim-snippets</li>
    <li>vimPlugins.vim-visual-multi</li>
    <li>vimPlugins.vim-vsnip</li>
    <li>vimPlugins.vimspector</li>
    <li>vimPlugins.vimtex</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 106592` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>spacevim</li>
    <li>vimPlugins.ale</li>
    <li>vimPlugins.barbar-nvim</li>
    <li>vimPlugins.brainfuck-vim</li>
    <li>vimPlugins.coc-nvim</li>
    <li>vimPlugins.defx-nvim</li>
    <li>vimPlugins.denite-nvim</li>
    <li>vimPlugins.deoplete-fish</li>
    <li>vimPlugins.deoplete-khard</li>
    <li>vimPlugins.deoplete-nvim</li>
    <li>vimPlugins.fern-vim</li>
    <li>vimPlugins.fzf-vim</li>
    <li>vimPlugins.julia-vim</li>
    <li>vimPlugins.lh-brackets</li>
    <li>vimPlugins.lh-vim-lib</li>
    <li>vimPlugins.nlua-nvim</li>
    <li>vimPlugins.nvim-gdb</li>
    <li>vimPlugins.nvim-lspconfig</li>
    <li>vimPlugins.nvim-lsputils</li>
    <li>vimPlugins.nvim-scrollview</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-refactor</li>
    <li>vimPlugins.nvim-ts-rainbow</li>
    <li>vimPlugins.popfix</li>
    <li>vimPlugins.tagbar</li>
    <li>vimPlugins.telescope-fzy-native-nvim</li>
    <li>vimPlugins.telescope-nvim</li>
    <li>vimPlugins.unite-vim</li>
    <li>vimPlugins.vim-clap</li>
    <li>vimPlugins.vim-floaterm</li>
    <li>vimPlugins.vim-illuminate</li>
    <li>vimPlugins.vim-lsp</li>
    <li>vimPlugins.vim-snipmate</li>
    <li>vimPlugins.vim-snippets</li>
    <li>vimPlugins.vim-visual-multi</li>
    <li>vimPlugins.vim-vsnip</li>
    <li>vimPlugins.vimspector</li>
    <li>vimPlugins.vimtex</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -1,5 +1,5 @@
 { lib, stdenv
-, python, cmake, meson, vim, ruby
+, python, gnumake, cmake, gcc, meson, vim, ruby
```
Comment (jlesquembre) [inline pkgs/misc/vim-plugins/overrides.nix:2]: Is this really necessary? gcc and make are part of the stdenv

File: pkgs/misc/vim-plugins/overrides.nix [hunk 2]
```diff
@@ -579,6 +579,20 @@ self: super: {
...
+  telescope-fzy-native-nvim = super.telescope-fzy-native-nvim.overrideAttrs(old: {
+    # it already contains precompiled libraries
+    # rm them and build our own
+    preFixup = ''
+      cd deps/fzy-lua-native
+      rm -rf static/*
+      ${gnumake}/bin/make
+      '';
+
+    buildInputs = [ gcc ];
```
Comment (jlesquembre) [inline pkgs/misc/vim-plugins/overrides.nix:591]: I think you can remove this line

File: pkgs/misc/vim-plugins/overrides.nix [hunk 3]
```diff
@@ -579,6 +579,20 @@ self: super: {
     dependencies = with super; [ bclose-vim ];
   });
 
+  telescope-fzy-native-nvim = super.telescope-fzy-native-nvim.overrideAttrs(old: {
+    # it already contains precompiled libraries
+    # rm them and build our own
+    preFixup = ''
```
Comment (jlesquembre) [inline pkgs/misc/vim-plugins/overrides.nix:585]: `make` should be invoked in the the build phase, it looks a bit strange to call it in the fixup phase: https://nixos.org/manual/nixpkgs/stable/#ssec-fixup-phase

**Review by jlesquembre — CHANGES_REQUESTED** [review]

**Comment by jlesquembre** [comment]
I didn't see your PR and created a similar one:
https://github.com/NixOS/nixpkgs/pull/108264

**Comment by onsails** [comment]
> I didn't see your PR and created a similar one:
> #108264
> 
> Some remarks:
> 
>     * Usually is better to squash your commits when you do an update
> 
>     * In the vim-plugins/generated.nix file, if you want to add only one package, is better to add to commit only that package. That reduce the possibility of a merge conflict

hm, I see, thanks for your review, good to know for my future PRs!

**Pull request closed without merging**